### PR TITLE
Fix CircleCI nightly run

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Currently on `main`, it always gives `empty` as jobs.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -82,22 +82,49 @@ jobs:
...
+            # We could not pass a nested dict, which is why we create the test_file_... parameters for every single job
+
+            - store_artifacts:
+                path: test_preparation/transformed_artifacts.json
+            - store_artifacts:
+                path: test_preparation/artifacts.json
             - continuation/continue:
-                  configuration_path: test_preparation/generated_config.yml
+                parameters:  test_preparation/transformed_artifacts.json
+                configuration_path: test_preparation/generated_config.yml
```
Comment (ydshieh) [inline .circleci/config.yml:127]: Just copied from `- fetch_tests` with a single difference 

> run: python utils/tests_fetcher.py | tee tests_fetched_summary.txt

becomes

> run: python utils/tests_fetcher.py --fetch_all | tee tests_fetched_summary.txt

File: utils/tests_fetcher.py [hunk 2]
```diff
@@ -1193,9 +1193,9 @@ def create_test_list_from_filter(full_test_list, out_path):
         default=None,
     )
     parser.add_argument(
-        "--commit_message",
```
Comment (ydshieh) [inline utils/tests_fetcher.py:1196]: this is never used.

(The actual used one is `commit_message = repo.head.commit.message`)

File: utils/tests_fetcher.py [hunk 3]
```diff
@@ -1212,6 +1212,9 @@ def create_test_list_from_filter(full_test_list, out_path):
             quit()
         if commit_flags["no_filter"]:
             print("Running all tests fetched without filtering.")
+
+        if args.fetch_all:
+            commit_flags["test_all"] = True
```
Comment (ydshieh) [inline utils/tests_fetcher.py:1217]: might not be perfect to do this, but it's a minimal change to make it work

**Review by ArthurZucker — APPROVED** [review]
Thanks for cleaning up after my mess!

**Pull request merged**